### ISSUE TITLE
React Native application wss connection not success

### ISSUE THREAD

**Issue by schavaLogi** [apollographql/subscriptions-transport-ws#352]
Hi , When i run RN application in mobile with our remote debug option websocket connection not established but with chrome debug it works.  url is "wss"

configuration:
"apollo-link": "^1.0.7",
"apollo-link-ws": "^1.0.4",
"subscriptions-transport-ws": "^0.9.5",
"react-native": "0.52.0",

Please help me to identify the diffrence in Chrome Debug and Normal mode

**Comment by perrosnk** [comment]
Is this on iOS or android?

**Comment by schavaLogi** [comment]
Both platforms we observed this issue. After detail analysis, we observed first we get disconnected event and then it will try for re-connect and will get connected successfully.

**Comment by oiojin831** [comment]
@schavaLogi Do't you have any problem using subscription in react native?
I got disconnected when I go background and come back to foreground

**Comment by schavaLogi** [comment]
Yes in iOS we have that problem that expected as per OS. in android it waits until server or load balancer closes connection.

The problem i was talking is mainly the initial connection is not smooth it randomly fails even when app is in foreground

**Comment by perrosnk** [comment]
@schavaLogi Any updated on this? Have you solved it?

**Comment by schavaLogi** [comment]
Yes, i had to mention wsTimeOut as 30sec as the KA packet reaching very late initially we had set as 8 sec.
